clients/v1_repl/v1_repl.go

```diff
@@ -297,6 +453,44 @@ func (c *Client) runAndShowQuery(query string) {
 	displayFunc(response)
 }
 
+// This function generates the prompt autocompletions
+func (c *Client) completer(d prompt.Document) []prompt.Suggest {
```

Comment:
Nitpick - is it possibly to arrange that `use` comes first in the completions if the user hasn't picked a database, and `SELECT` and `INSERT` are the first completions if the user has picked a database (followed by the non-query functions like `clear`, `format`, etc)

Reply:
Also after `SELECT` it would be cool to turn off the suggestions until after the `FROM` - it's difficult to suggest field names and the `ALL` `ANY` etc get distracting.

Reply:
We already have the `influxql` repo that has a parser implemented, I wonder how hard it would be to add a feature there to return an intermediate parse result that hints at the actual possible completions instead of giving all the possible InfluxQL keywords even when they are not relevant?

Reply:
Looks like these lines here implement the logic of the parser: https://github.com/influxdata/influxql/blob/1ba470371ec093d57a726b143fe6ccbacf1b452b/parse_tree.go#L102-L235

I'm not sure if I can use that code directly, but I can structure the autocompletion with the same logic if that makes sense.

Reply:
I'm a bit worried about maintenance here - I don't want to end up with two implementations of the InfluxQL parser.

Reply:
I added made a new commit to the PR that handles autocompletion a much more extensible way. You may see it as another implementation of the influxql parser, but the builtin REPL commands combined with the deprecated influxql commands in 2.x make it different enough to me that it warrants this change. Let me know what you think.

Reply:
Also, I've got an idea for helping guide users through deprecated queries. I can add a feature to the go-prompt library to allow grayed-out or yellow options signifying deprecated commands. Those options wouldn't be selectable, and they would always appear after non-deprecated options. What do you think of that idea in terms of resolving #397 ?

Reply:
I think I'd prefer to just not complete the non-implemented InfluxQL at all - it's possible we would implement more of the old InfluxQL spec in the future, so I don't want to be too aggressive. `https://docs.influxdata.com/influxdb/v2.2/query-data/influxql/#` gives the implemented vs. non-implemented subset.

Also it would be nice (though not critical) if the REPL was backward-compatible to old 1.x databases.

Maybe we could enhance the messages from the database itself for old queries? It is already parsing the influxql statements, it would be easy enough to say something like `CREATE DATABASE is a 1.x DML statement that is not supported` direct from the 2.x API, and then the client can just report the error it gets back.